test_tools/src/test_tools/learn_objects.py

```diff
@@ -0,0 +1,73 @@
+#! /usr/bin/env python
+
+import sys
+
+import rospy
+import PyKDL as kdl
+
+
+from robot_smach_states.util.designators import EntityByIdDesignator
+from robot_skills.util.kdl_conversions import VectorStamped
+from robot_skills.util.robot_constructor import robot_constructor
+
+
+def look_at_entity(robot, location_des):
+    # look on top of the surface
+    entity = location_des.resolve()
+    area = "on_top_of"
+    search_volume = entity.volumes[area]
+    x_obj = 0.5 * (search_volume.min_corner.x() + search_volume.max_corner.x())
```

Comment:
@LoyVanBeek is there a smarter/more efficient way to take the average value or is this the way to go?

Reply:
A `Volume` has a `center_point` (useage her: `search_volume.center_point`). It uses the exact same calculation, also for the Z, while you just take the `min` here. But: no, I don't knwo a more efficient way.